src/Sylius/Bundle/ApiBundle/Serializer/CustomerDenormalizer.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Serializer;
+
+use DateTimeInterface;
+use Sylius\Calendar\Provider\DateTimeProviderInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerAwareTrait;
+
+/** @experimental */
+final class CustomerDenormalizer implements ContextAwareDenormalizerInterface, DenormalizerAwareInterface
+{
+    use DenormalizerAwareTrait;
+
+    private const ALREADY_CALLED = 'sylius_customer_denormalizer_already_called';
+
+    public function __construct(private DateTimeProviderInterface $calendar)
+    {
+    }
+
+    public function supportsDenormalization(mixed $data, string $type, string $format = null, array $context = []): bool
+    {
+        return
+            !isset($context[self::ALREADY_CALLED]) &&
+            is_array($data) &&
+            is_a($type, CustomerInterface::class, true)
+        ;
+    }
+
+    public function denormalize(mixed $data, string $type, string $format = null, array $context = [])
+    {
+        $context[self::ALREADY_CALLED] = true;
+        $data = (array) $data;
+
+        $user = $data['user'] ?? null;
+        if (null !== $user && array_key_exists('verified', $user)) {
+            $data['user']['verified'] = true === $user['verified'] ? $this->calendar->now()->format(DateTimeInterface::RFC3339) : null;
```

Comment:
Shouldn't the date be set to the `verifiedAt` field?

Reply:
No, because I used a `serialized-name`. More information [here](https://github.com/Sylius/Sylius/pull/15457#discussion_r1373544721).